llvm/include/llvm/MC/MCSection.h

```diff
@@ -431,16 +428,38 @@ class MCDataFragment : public MCEncodedFragment {
 ///
 class MCRelaxableFragment : public MCEncodedFragment {
   /// The instruction this is a fragment for.
-  MCInst Inst;
```

Comment:
Is it ok to always drop Flags? We might end up with a different encoding for the relaxed instruction.

Reply:
X86MCInstLower::Lower calls setFlags and with allowEnhancedRelaxation, these can end up in a relaxable fragment?

Reply:
You are right. I take back my previous comment.. Updated the description

> Unfortunately, we also have to encode MCInst::Flags to support the EVEX prefix, e.g. `{evex} xorw $foo, %ax`.

I only noticed the case after adding the `getFlags() == 0` assertion... The #78545 case should probably switch to llvm-objdump -dr